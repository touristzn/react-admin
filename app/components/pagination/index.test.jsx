import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Pagination from './index'

let wrapper

describe('<Pagination />', () => {
  const props = {
    className: 'mt-20',
    currentChange: jest.fn(),
    currentPage: 1,
    pageSize: 20,
    total: 140,
  }

  beforeEach(() => {
    wrapper = mount(
      <Pagination {...props} />
    )
  })

  it('the custom class', () => {
    expect(wrapper.props().className).toContain(props.className)
  })

  it('click the page number should change the active stauts and trigger the function', () => {
    wrapper.find('.page li').at(3).simulate('click')
    // 元素索引是从0开始，所以要加1
    wrapper.setProps({ currentPage: 4 })
    expect(wrapper.find('.page li').at(3).props().className).toBe('active')
    expect(props.currentChange).toHaveBeenCalledWith(4)
  })

  it('up and down bottons', () => {
    wrapper.setProps({ currentPage: 1 })
    expect(wrapper.find('.btn-prev').props().className).toContain('disabled')
    // 总页数: props.total / props.pageSize
    wrapper.setProps({ currentPage: 7 })
    expect(wrapper.find('.btn-next').props().className).toContain('disabled')
  })

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})