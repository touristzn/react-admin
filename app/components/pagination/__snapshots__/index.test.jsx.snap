// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> renders correctly 1`] = `
<Component
  className="mt-20"
  currentChange={
    [MockFunction] {
      "calls": Array [
        Array [
          4,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  currentPage={1}
  pageSize={20}
  total={140}
>
  <Pagination
    __={[Function]}
    className="mt-20"
    currentChange={
      [MockFunction] {
        "calls": Array [
          Array [
            4,
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    currentPage={1}
    pageSize={20}
    total={140}
  >
    <div
      className="pagination mt-20"
    >
      <span
        className="pagination__total"
      >
        共
         
        140
         
        条
      </span>
      <div
        className="pagination__sizes"
      >
        <button
          className="btn-prev disabled"
          onClick={[Function]}
        >
          上一页
        </button>
        <ul
          className="page"
        >
          <li
            className="active"
            onClick={[Function]}
          >
            1
          </li>
          <li
            className=""
            key="2"
            onClick={[Function]}
          >
            2
          </li>
          <li
            className=""
            key="3"
            onClick={[Function]}
          >
            3
          </li>
          <li
            className=""
            key="4"
            onClick={[Function]}
          >
            4
          </li>
          <li
            className=""
            key="5"
            onClick={[Function]}
          >
            5
          </li>
          <li
            className=""
            key="6"
            onClick={[Function]}
          >
            6
          </li>
          <li
            className=""
            onClick={[Function]}
          >
            7
          </li>
        </ul>
        <button
          className="btn-next "
          onClick={[Function]}
        >
          下一页
        </button>
      </div>
    </div>
  </Pagination>
</Component>
`;
